vars/crossClusterPromote.groovy

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env groovy
+
+class CopyImageInput implements Serializable{
+	String sourceImageName
+	String sourceImagePath
+	String sourceImageTag = "latest"
+	String destinationImageName
+	String destinationImageTag
+	String destinationImagePath
+	String targetRegistryCredentials = "other-cluster-credentials"
+	String clusterUrl = ""
+	String clusterToken = ""
+
+    CopyImageInput init(){
+		if(!destinationImageName?.trim()) destinationImageName = sourceImageName
+        if(!destinationImageTag?.trim()) destinationImageTag = sourceImageTag
+        if(!destinationImagePath?.trim()) destinationImagePath = sourceImagePath
+        return this
+    }
+}
+
+def call(Map input){
+	call(new CopyImageInput(input).init())
+}
+
+def call(CopyImageInput input) {
+	openshift.withCluster(input.clusterUrl, input.clusterToken) {
+		def localToken = readFile("/var/run/secrets/kubernetes.io/serviceaccount/token").trim()
+
+		def secretData = openshift.selector("secret/${input.targetRegistryCredentials}").object().data
+		def registry = sh(script:"set +x; echo ${secretData.registry} | base64 --decode", returnStdout: true)
+		def token = sh(script:"set +x; echo ${secretData.token} | base64 --decode", returnStdout: true)
+		def username = sh(script:"set +x; echo ${secretData.username} | base64 --decode", returnStdout: true)
+
+		openshift.withProject("${input.sourceImagePath}") {
+			def localRegistry = openshift.selector( "is", "${input.sourceImageName}").object().status.dockerImageRepository
+			def from = "docker://${localRegistry}:${input.sourceImageTag}"
+			def to = "docker://${registry}/${input.destinationImagePath}/${input.destinationImageName}:${input.destinationImageTag}"
+
+			echo "Now Promoting ${from} -> ${to}"
+			sh """
+				set +x
+				skopeo copy --remove-signatures \
```

Comment:
@syvanen one thing to look into... the openshift team has added native functionality for moving images between regstries, which we might want to look into to replace skopeo. Take a look at `oc image mirror`.

Reply:
@etsauer That is a good point but last time I checked it didn't use the openshift authentication at all. Instead it requires Docker authentication `docker login` to function. From the help for `oc image mirror`:

> The default docker credentials are used for authenticating to the registries. 

So I'm not sure how that would work in our Jenkins slave images.

Reply:
@syvanen adding to @etsauer comments, you may want to check out https://github.com/openshift/jenkins/issues/632 as well.

Reply:
thanks @etsauer and @Gl4di4torRr, I'll test the copying with that and update this PR to use that method.